### ISSUE TITLE
In-Game: sprite_get_uvs() returns data of previous image_index if manually incrementing the value by 0.3 each time and reaching 3.0

### ISSUE THREAD

**Issue by dt-universe** [YoYoGames/GameMaker-Bugs#2601]
### Description

If I use instance variable which I increment on 0.3 each frame, when it will be equal to 3, sprite_get_uvs will return data of 2 frame, but GameMaker reports that variable is exactly 3 with string() or string_format(), draw_sprite_ext draws 3 frame, comparison with 3 also returns true.

I want to maintain that this issue is old, because originally I found it on 2022.11.1.75 runtime. Maybe other functions have the same issue, but who knows.

### Expected Change

sprite_get_uvs should work the same with numbers as other functions.

### Steps To Reproduce

You can check my sample project and you will see that sprite_get_uvs returns data of 2 frame, but draw_sprite_ext draws 3 frame and other outputs report that variable is 3.

```
CREATE | Image index: 3.0000000000 | UVS data: [ 0.27,0.01,0.52,0.26,0,0,1,1 ]
DRAW | Image index: 3.0000000000 | UVS data: [ 0.01,0.27,0.26,0.52,0,0,1,1 ] | Is variable equal 3? 1
```
[sprite_get_uvs_package.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13775890/sprite_get_uvs_package.zip)

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2023.1100 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by YYDan** [comment]
I am not sure which runtime version "2025.11.1.75" was meant to be ;) but your sample does still reproduce the problem for me in the current IDE v2023.11.1.129  Runtime v2023.11.1.160 anyway...

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/15669932/424fbabd-b7e2-4908-9dc7-14610da20f4b)

Here's my version of the test project: [sprite_get_uvs_issue.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13778294/sprite_get_uvs_issue.zip)

**Comment by YYDan** [comment]
Removed the math_set_epsilon() guess comment, as the sample doesn't show this.

**Comment by dt-universe** [comment]
> I am not sure which futuristic runtime version "2025.11.1.75" was meant to be ;)

Oh, I meant 2022.11.1.75.

> Removed the math_set_epsilon() guess comment, as the sample doesn't show this.

It was my guess after submitting and this fixed "third" frame.

**Comment by JonnoRobson** [comment]
This is unlikely to be changed as it is an issue with floating point precision when passing a non-integer real value into a built in function when that function wants a whole integer value. `sprite_get_uvs` really wants subimage to be an integer value as anything other than a whole integer value does not make sense (e.g. what does a subimage index of 0.6 actually mean when getting sprite uvs?)

**Comment by JonnoRobson** [comment]
The discrepancy between the result of `sprite_get_uvs` and `draw_sprite_ext` was the result of a difference in how the subimage argument of the functions was handled. 

Fixed by changing the handling of the subimage argument in `sprite_get_uvs` to match how it is handled in `draw_sprite_ext`. Also made the same fix in `sprite_get_texture` as both functions had the same issue.

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.200.0.454  Runtime v2024.200.0.466

**Comment by dt-universe** [comment]
Hello! I'm still on v2022 and sometimes this issue happens, epsilon doesn't help, because if image_index is something like 0.99999999999999, draw_sprite_ext draws 0 image_index, but if image_index is 2.9999999999999, then it draws 3 image_index. May you say, how can I temporary fix it on my version on GML side?
